firefox-ios/Storage/Rust/RustRemoteTabs.swift

```diff
@@ -30,19 +31,20 @@ public class RustRemoteTabs {
         self.logger = logger
 
         queue = DispatchQueue(label: "RustRemoteTabs queue: \(databasePath)", attributes: [])
+        commandQueue = DispatchQueue(label: "RustRemoteTabs command queue", attributes: [])
     }
 
     private func open() -> NSError? {
         store = TabsStore(path: databasePath)
         isOpen = true
-        commandQueue = RemoteTabsCommandQueue()
-        commandQueue?.openCommandStore(tabsStore: store!)
+        tabsCommandQueue = RemoteTabsCommandQueue(tabsStore: store!)
         return nil
     }
 
     private func close() -> NSError? {
         store = nil
         isOpen = false
+        tabsCommandQueue = nil
```

Comment:
I suggested dropping `tabsCommandQueue` above, but on a more general note, there's no need to null out queues: if there are any work items scheduled on a queue, GCD will keep it alive, and keep running those work items to completion; nulling it out doesn't prevent any operations that are already scheduled on it from running.

Reply:
There are a lot of queues around. The `tabsCommandQueue` is actually the queue being used to store the commands not the dispatch queue.

Reply:
Oh gosh, I'm sorry I completely mixed them up here! 🫣